test/module/irohad/ametsuchi/postgres_query_executor_test.cpp

```diff
@@ -676,29 +676,62 @@ namespace iroha {
       using Impl = QueryTxPaginationTest;
 
       // create valid transactions and commit them
-      void createTransactionsAndCommit(size_t transactions_amount) {
+      void createTransactionsAndCommit(size_t transactions_amount,
+                                       bool build_blocks = false) {
         addPerms(Impl::getUserPermissions());
-
         auto initial_txs = Impl::makeInitialTransactions(transactions_amount);
         auto target_txs = Impl::makeTargetTransactions(transactions_amount);
-
+        auto size_diff = initial_txs.size();
         tx_hashes_.reserve(target_txs.size());
         initial_txs.reserve(initial_txs.size() + target_txs.size());
         for (auto &tx : target_txs) {
           tx_hashes_.emplace_back(tx.hash());
           initial_txs.emplace_back(std::move(tx));
         }
-
-        auto block = createBlock(initial_txs, 1);
-
-        apply(storage, block);
+        if (build_blocks) {
+          if (size_diff != 0) {
+            auto first_tx = std::vector<shared_model::proto::Transaction>(
+                initial_txs.begin(), initial_txs.begin() + size_diff + 1);
+            auto first_block = createBlock(first_tx, 1);
+            apply(storage, first_block);
+            for (size_t i = size_diff + 1; i < initial_txs.size(); i++) {
+              auto current_tx =
+                  std::vector<shared_model::proto::Transaction>{initial_txs[i]};
+              auto current_block = createBlock(current_tx, i);
+              apply(storage, current_block);
+            }
+          } else {
+            for (size_t i = size_diff; i < initial_txs.size(); i++) {
+              auto current_tx =
+                  std::vector<shared_model::proto::Transaction>{initial_txs[i]};
+              auto current_block = createBlock(current_tx, i + 1);
+              apply(storage, current_block);
+            }
```

Comment:
The loop is the same here and in lines [697, 702] - why not to extract function/method?